musicbrainz/csv/convert_to_csv.py

```diff
@@ -0,0 +1,101 @@
+# This file converts a JSON Lines file from MusicBrainz to a CSV file
+# For OpenRefine Reconciling
+# Input file can be downloaded in
+# https://data.metabrainz.org/pub/musicbrainz/data/json-dumps/20240522-001002/
+# Output file can be imported to OpenRefine
+import json
+import copy
+
+header = ["id"]
+values = []
+
+with open("test2", "r") as f:
+    json_data = [json.loads(m) for m in f]
+
+# print(json_data)
+
+
+def extract(data, first_level: bool = True, key: str = "", value: dict = {}):
+    if key != "":
+        first_level = False
+    if "relations" in key or "aliases" in key:
+        # ignore relations and aliases to make output simplier
+        return
+
+    if isinstance(data, dict):
+        if first_level:
+            global values
+            for k in data:
+                extract(data[k], False, k, value)
+            values.append(copy.deepcopy(value))
+            value = {}
+
+        else:
+            for k in data:
+                if k == "id":
+                    extract(data["id"], first_level, key + "_id", value)
+
+                if k == "name":
+                    extract(data["name"], first_level, key + "_name", value)
+
+                if isinstance(data[k], dict) or isinstance(data[k], list):
+                    extract(data[k], first_level, key + "_" + k, value)
+
+    elif isinstance(data, list):
+        rep_count = 0
+        for element in data:
+            rep_count += 1
+            if isinstance(element, dict) and rep_count <= 3:
+                if first_level:
+                    rep_count = 0
+                    extract(element, first_level, key, value)
+                else:
+                    extract(element, first_level, key + str(rep_count), value)
+
+    else:
+        global header
+        if key not in header:
+            header.append(key)
+
+        if isinstance(data, str):
+            data = data.replace(",", "")
+            data = data.replace("\r\n", "")
+
+        if data == None:
+            data = ""
+
+        value[key] = data
+        return
```

Comment:
This function always returns `None`?

Reply:
Yes, but I think this can be optimized.

Reply:
It adds all the data into a global variable. Do recursive returns make it better?

Reply:
I'm not necessarily looking for optimized....I'm just trying to understand what this function does. It is hard for me to understand the way it is written.... more comments maybe? Or some discrete functions?

Reply:
I will add comments.